utils/table.lua

```
@@ -32,21 +23,11 @@ table.add_all = function(t1, t2)
     end
 end
 
-table.size = function(t)
-    assert_argument_valid(t)
-    local size = 0
-    for _ in pairs(t) do
-        size = size + 1
-    end
-    return size
-end
-
 --- Checks if a table contains an element
 -- @param t table to search
 -- @param e element to search for
 -- @returns the index of an element or -1
 table.index_of = function(t, e)
-    assert_argument_valid(t)
```

Shouldn't the asserts be replaced?

---

Be replaced as in be used?
I asked Val if asserting was computationally costly and he indicated they were. Because we suppress errors outside of debug anyways, it doesn't seem beneficial to spend computation time asserting arguments.

---

Fair enough! It's better to have this kind of validation only when it's not relying on performance.